src/components/SuggestionsList/SuggestionsList.tsx

```diff
@@ -0,0 +1,158 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import '../../yamui';
+import * as React from 'react';
+import { NestableBaseComponentProps } from '../../util/BaseComponent/props';
+import SuggestionsListItem, { SuggestionItem } from './SuggestionsListItem';
+import Spinner from '../Spinner';
+import Text, { TextSize, TextColor } from '../Text';
+import Callout from '../Callout';
+import Block, { GutterSize } from '../Block';
+import './SuggestionsList.css';
+
+export interface SuggestionsListProps extends NestableBaseComponentProps {
+  /**
+   * The target that the Callout should try to position itself based on.
+   */
+  target: HTMLElement;
+  /**
+   * The active search that produced the current state.
+   */
+  searchText: string;
+  /**
+   * Text to show next to the loading spinner.
+   */
+  loadingText: string;
+  /**
+   * Text to show when the results are empty for the active search.
+   */
+  noResultsText: string;
+  /**
+   * Flag to indicate that a search is in flight
+   * @default false
+   */
+  isLoading?: boolean;
+  /**
+   * The groups of items to render.
+   */
+  groupedItems?: SuggestionItemGroupProps[];
+  /**
+   * The index of current keyboard selection.
+   */
+  selectedId?: string | number;
+  /**
+   * Called when an item in a group is selected by clicking.
+   */
+  onItemSelected(item: SuggestionItem): void;
```

Comment:
is there any value returning the entire SuggestionItem instead of its id? the item object is only presentational (id, name, description, imageUrl), could the outer component ever use the full item?

this seems like a nitpicky question, but in general we're trying to pass around specific values rather than bags of things, and we made the decision with Dropdown to only return the id instead of the full item:
https://github.com/Microsoft/YamUI/blob/master/src/components/Dropdown/Dropdown.tsx#L81

Reply:
but then the user has to look the data up? I wouldnt characterize it as "a bags of things" because you a) pass by object reference b) this is TypeScript so its well defined whats inside

Reply:
Whatever we do, we should be consistent across components. If `Dropdown` is already doing things one way, either we do the same, or we change both.

I'm fine with just passing the `id`. It's similar to how a plain `select` would behave, where you'd get the `value` of the selected `option`.

Reply:
the user always has to look data up because this item is only the visual data (id, name, description, imageUrl), it's not the actual data object managed by the application. in practice, your code will still use selectedItem.id to look up the actual data